activerecord/test/cases/persistence_test.rb

```diff
@@ -924,7 +924,7 @@ def test_save_touch_false
       assert_equal instance.created_at, created_at
       assert_equal instance.updated_at, updated_at
     ensure
-      ActiveRecord::Base.connection.drop_table :widgets
+      ActiveRecord::Base.connection.drop_table widget.table_name
```

Comment:
Instead of changing the table name to prevent the collision, we should use `widget.reset_column_information`. The thing is that this method clears the table cache, which is connection wide.